llama_index/callbacks/base.py

```diff
@@ -88,3 +113,36 @@ def remove_handler(self, handler: BaseCallbackHandler) -> None:
     def set_handlers(self, handlers: List[BaseCallbackHandler]) -> None:
         """Set handlers as the only handlers on the callback manager."""
         self.handlers = handlers
+
+    @contextmanager
+    def as_trace(self, run_id: str) -> Generator[None, None, None]:
+        """Context manager tracer for lanching and shutdown of runs."""
+        self.launch(run_id=run_id)
+        yield
+        self.shutdown(run_id=run_id)
+
+    def launch(self, run_id: Optional[str] = None) -> None:

```

Comment:
nit: can we just call it `start_trace` / `end_trace` instead of `launch` / `shutdown`?

Reply:
yea that's cleaner

Reply:
nit: also can we call it `trace_id` to reduce the number of concepts? (unless run and trace is not the same thing)

Reply:
Done! using `trace` terminology everywhere now